include/mqtt/endpoint.hpp

```diff
@@ -669,13 +670,14 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
 
     // Original handlers
 
+protected:
     /**
      * @brief Close handler
      *
      * This handler is called if the client called `disconnect()` and the server closed the socket cleanly.
      * If the socket is closed by other reasons, error_handler is called.
      */
-    virtual void on_close() noexcept = 0;
+    virtual void on_close() noexcept {}
```

Comment:
Instead of making this an empty implementation, i would rather put an empty implementation into client.hpp.

Reply:
server uses callable_overlay like https://github.com/redboltz/mqtt_cpp/blob/master/include/mqtt/server.hpp#L45

The empty definition is for them.
Not related to the `client`.

Reply:
I see.

I guess I would suggest making a new server.hpp to provide the empty implementations.

This would also allow us to move some server-side only functionality out of endpoint.hpp, into server.hpp (not in this PR, of course).

But if you don't want to I guess it's not important.

Reply:
See https://wandbox.org/permlink/5Gi37TYU9awCyv4C

Reply:
Right, I understood the problem.

My suggestion was to keep endpoint.hpp with pure-virtual, and add new file "server.hpp", which inherits from endpoint, and provides empty override for the pure-virtual functions, and then callable_overlay uses server.hpp, instead of endpoint.

Reply:
Ok, I will do this.